### ISSUE TITLE
Crash when playing Ultracube

### ISSUE THREAD

**Issue by PFQNiet** [ClaudeMetz/FactoryPlanner#240]
### Problem Description

Playing the Ultracube mod, I was preparing a factory and wanted to select Steam to find out how much Water I needed in total for a plan, when the mod crashed:

```
The mod Factory Planner (1.1.78) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::Custom event (ID 232)
__factoryplanner__/backend/data/Machine.lua:62: attempt to index field '?' (a nil value)
stack traceback:
	__factoryplanner__/backend/data/Machine.lua:62: in function 'normalize_fuel'
	__factoryplanner__/backend/data/Line.lua:100: in function 'change_machine_to_proto'
	__factoryplanner__/backend/data/Line.lua:151: in function 'change_machine_to_default'
	__factoryplanner__/ui/dialogs/recipe_dialog.lua:94: in function 'attempt_adding_line'
	__factoryplanner__/ui/dialogs/recipe_dialog.lua:258: in function 'early_abort'
	__factoryplanner__/ui/base/modal_dialog.lua:154: in function 'enter'
	__factoryplanner__/ui/event_handler.lua:209: in function <__factoryplanner__/ui/event_handler.lua:189>
	[C]: in function 'raise_event'
	__factoryplanner__/util/raise.lua:20: in function 'open_dialog'
	__factoryplanner__/ui/main/item_boxes.lua:11: in function 'add_recipe'
	__factoryplanner__/ui/main/item_boxes.lua:161: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
stack traceback:
	[C]: in function 'raise_event'
	__factoryplanner__/util/raise.lua:20: in function 'open_dialog'
	__factoryplanner__/ui/main/item_boxes.lua:11: in function 'add_recipe'
	__factoryplanner__/ui/main/item_boxes.lua:161: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
```

### Reproduction

1. With Ultracube mod enabled, start a new game and enter `/editor` mode.
2. Open tech tree, select the green tech card and insta-research it.
3. Create a new Factory Planner, select Basic Matter Unit. Toggle unresearched recipes to select the replicator-based recipe.
4. Recipe should show Matter Replication Gel and Steam. Click on the Steam to (try to) add the Water boiler.
5. Crash!

**Comment by ClaudeMetz** [comment]
Thanks, waiting on the 2.0 version of ultracube to verify whether it's still an issue.

**Comment by ClaudeMetz** [comment]
I looked into this, and the issue is fixed/the situation changed with 2.0. There are more temperature-related changes coming, but I'll consider this solved for now.
